frontend/src/lib/components/SmartNodeSelector/smartNodeSelector.tsx

```diff
@@ -952,29 +960,34 @@ export class SmartNodeSelectorComponent extends React.Component<SmartNodeSelecto
 
     protected updateSelectedTagsAndNodes(initialUpdate = false): void {
         const { onChange, maxNumSelectedNodes } = this.props;
-        const selectedTags: string[] = [];
+        const selectedTags: SmartNodeSelectorTag[] = [];
         const selectedNodes: string[] = [];
         const selectedIds: string[] = [];
         loop1: for (let i = 0; i < this.countTags(); i++) {
             const nodeSelection = this.nodeSelection(i);
             if (nodeSelection.getCompleteNodePathAsString() !== "") {
-                selectedTags.push(nodeSelection.getCompleteNodePathAsString());
+                selectedTags.push({
+                    text: nodeSelection.getCompleteNodePathAsString(),
+                    isValid: nodeSelection.isValid(),
+                });
             }
             if (nodeSelection.isValid() && !this.checkIfSelectionIsDuplicate(nodeSelection, i)) {
                 const matchedNodePaths = nodeSelection.exactlyMatchedNodePaths();
+                const matchedIds = nodeSelection.exactlyMatchedNodeIds();
                 for (let j = 0; j < matchedNodePaths.length; j++) {
                     if (selectedNodes.length >= maxNumSelectedNodes && maxNumSelectedNodes > 0) {
                         break loop1;
                     }
                     selectedNodes.push(matchedNodePaths[j]);
-                    selectedIds.push(nodeSelection.getId() || "");
+                    selectedIds.push(matchedIds[j][matchedIds[j].length - 1] ?? "");
                 }
             }
         }
         if (
             !this.selectedNodes ||
             selectedNodes.length !== this.selectedNodes.length ||
-            JSON.stringify(this.selectedNodes) !== JSON.stringify(selectedNodes)
+            JSON.stringify(this.selectedNodes) !== JSON.stringify(selectedNodes) ||
+            JSON.stringify(this.props.selectedTags) !== JSON.stringify(selectedTags)
```

Comment:
Optionally, could use lodash's [`isEqual()`](https://lodash.com/docs/4.17.15#isEqual) here

Reply:
Perhaps true. What do you think @rubenthoms ?

Reply:
`SmartNodeSelector` comes from a repository (wcc) where we did not have `lodash` as a dependency. That's why it hasn't been used in the component so far. Since we have our own implementation copy of it in the Webviz NG repository (due to Tailwind styling), we could of course start replacing all these occurrences with `isEqual` as it is generally more performant. However, since we might (although with low probability) apply the fixes from Webviz in wcc, it seems beneficial to have both implementations use the same code for now. Either we stick to the no-lodash variant until all ties to wcc are really cut or we refactor both versions to using `isEqual`. This, however, does not seem to be one of the important tasks at the moment as the lower performance has not caused any issues so far.

Reply:
We'll leave it as it is for now :)